tests/main/testlib-defer/task.yaml

```diff
@@ -0,0 +1,75 @@
+summary: tests can use reliable defer logic
+detail: |
+    The defer and run_deferred shell functions allow to closely couple commands
+    with any necessary cleanup. For details see the documentation of the
+    functions in the defer.sh file below.
+execute: |
+    #shellcheck source=tests/lib/defer.sh
+    . "$TESTSLIB/defer.sh"
+
+    # Actions can be deferred.
+    for i in 1 2 3; do
+        touch "file-$i"
+        defer rm "file-$i"
+        test -f "file-$i"
+    done
+
+    # Internally the state is stored in the following files
+    test -f .auto_defer_count
+    test -f .auto_defer_0
+    test -f .auto_defer_1
+    test -f .auto_defer_2
+    test -f .defer_lock
+
+    MATCH 2 < .auto_defer_count
+    MATCH 'rm file-1' < .auto_defer_0
+    MATCH 'rm file-2' < .auto_defer_1
+    MATCH 'rm file-3' < .auto_defer_2
+
+    # Deferred actions can be invoked as seen here.
+    run_deferred
```

Comment:
i understand needing to do this manually for the test, but `defer.sh` could do `trap run_deferred RETURN`, couldn't it?

Reply:
I did not consider `RETURN` but my main motivation was to allow it to be used explicitly in prepare/restore section (for `defer` and `run_deferred`, respectively) which requires explicit action in absence of additional information about the spread `verb`.

Can you explain how using `RETURN` might improve usability please?

Reply:
s/RETURN/EXIT/ ? I'm not familiar with RETURN trap, is that a Bash specific feature?

Reply:
Hah, it is. I'm fine with that. Spread executes tests under bash, so making the defer thing more bash oriented would make no harm.